api/src/main/java/org/openmrs/module/openconceptlab/importer/Importer.java

```diff
@@ -171,13 +179,43 @@ public void run() {
 
 					@Override
 					public void run() throws IOException {
-						in = new CountingInputStream(inputStream);
-
-						importService.updateSubscriptionUrl(anImport, IMPORTED_VIA_API);
+						if (in != null) {
+							if (!in.equals(inputStream)) {
+								setInputStream(inputStream);
+							}
+							importService.updateSubscriptionUrl(anImport, IMPORTED_VIA_API);
+							//TODO Is this date OK?
+							Date date = new Date();

```

Comment:
Good question. I guess we need to stay with that unless we can find the date inside the file...